pkg/cmd/server/api/validation/validation_test.go

```diff
@@ -206,3 +211,41 @@ MCUXgckTpKCuGwbJk7424Nb8bLzf3kllAiA5mUBgjfr/WtFSJdWcPQ4Zt9KTMNKD
 EUO0ukpTwEIl6wIhAMbGqZK3zAAFdq8DD2jPx+UJXnh0rnOkZBzDtJ6/iN69AiEA
 1Aq8MJgTaYsDQWyU/hDq5YkDJc9e9DSCvUIzqxQWMQE=
 -----END RSA PRIVATE KEY-----`)
+
+func TestLDAPSyncConfigFixtures(t *testing.T) {
+	fixtures := []string{}
+
+	// build a list of common configurations for all schemas
+	schemas := []string{"rfc2307", "ad", "augmented-ad"}
+	for _, schema := range schemas {
+		fixtures = append(fixtures, schema+"/sync-config.yaml")
+		fixtures = append(fixtures, schema+"/sync-config-dn-everywhere.yaml")
+		fixtures = append(fixtures, schema+"/sync-config-partially-user-defined.yaml")
+		fixtures = append(fixtures, schema+"/sync-config-user-defined.yaml")
+	}
+
+	for _, fixture := range fixtures {
+		var config api.LDAPSyncConfig
+
+		yamlConfig, err := ioutil.ReadFile("./../../../../../test/extended/authentication/ldap/" + fixture)
+		if err != nil {
+			t.Errorf("could not read fixture at %q: %v", fixture, err)
+			continue
+		}
+
+		jsonConfig, err := yaml.ToJSON(yamlConfig)
+		if err != nil {
+			t.Errorf("could not convert YAML fixture at %q to JSON: %v", fixture, err)
+			continue
+		}
+
+		if err := runtime.DecodeInto(configapilatest.Codec, jsonConfig, &config); err != nil {
+			t.Errorf("could not deocde JSON fixture at %q into internal type: %v", fixture, err)
+			continue
+		}
+
+		if results := ValidateLDAPSyncConfig(&config); len(results.Errors) > 0 || len(results.Warnings) > 0 {
+			fmt.Errorf("validation of fixture at %q failed with %d warnings and %d errors:\n\twarnings:\n%v\n\terrors:\n%v", fixture, len(results.Errors), len(results.Warnings), results.Warnings, results.Errors)
```

Comment:
Good question. It currently passes locally, I'll reword this check to ignore warnings or expect that warning and determine why I'm not saying the warning right now.